api/repositories/shared.go

```diff
@@ -59,3 +67,73 @@ func getLabelOrAnnotation(mapObj map[string]string, key string) string {
 	}
 	return mapObj[key]
 }
+
+func GetNamespacedResourceNamespace(ctx context.Context, privilegedClient client.Client, userClientFactory UserK8sClientFactory, authInfo authorization.Info, resourceGUID string, resourceList client.ObjectList) (string, error) {
+	err := privilegedClient.List(ctx, resourceList, client.MatchingFields{"metadata.name": resourceGUID})
+	if err != nil { // untested
+		return "", fmt.Errorf("privileged list failed: %w", err)
+	}
+
+	switch t := resourceList.(type) {
+	case *workloadsv1alpha1.CFAppList:
+		err := CheckListLength(len(t.Items), "app")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *workloadsv1alpha1.CFPackageList:
+		err := CheckListLength(len(t.Items), "package")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *workloadsv1alpha1.CFBuildList:
+		err := CheckListLength(len(t.Items), "build")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *workloadsv1alpha1.CFProcessList:
+		err := CheckListLength(len(t.Items), "process")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *networkingv1alpha1.CFRouteList:
+		err := CheckListLength(len(t.Items), "route")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *networkingv1alpha1.CFDomainList:
+		err := CheckListLength(len(t.Items), "domain")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *servicesv1alpha1.CFServiceBindingList:
+		err := CheckListLength(len(t.Items), "service-binding")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	case *servicesv1alpha1.CFServiceInstanceList:
+		err := CheckListLength(len(t.Items), "service-instance")
+		if err != nil {
+			return "", err
+		}
+		return t.Items[0].Namespace, nil
+	default:
```

Comment:
This is obviously would not be a proper implementation - need to handle it in a better way.